### PR TITLE
Prepare for 0.0.1 release

### PR DESCRIPTION
**Pull request by sluongng**
In BzlMod, we have a top level module named `toolchains_buildbuddy`,
which contains the `buildbuddy` extension.
The extension will then be used to create the `buildbuddy_toolchain`
repository, which contains our toolchain and platform defintions.

Add automatic toolchains and exec platforms registration in the
MODULE.bazel of the `buildbuddy_toolchain` module. This means that any
repository which use our module would immediately get the relevant
toolchains and exec platforms available to them.

Fix documentations and comments in README.md and bzlmod examples.

### CODE REVIEW

**Comment by sluongng** [comment]
Marking this as a draft for now because we gonna need to iterate on this.

File: MODULE.bazel [hunk 1]
```diff
@@ -1,8 +1,24 @@
 module(
     name = "buildbuddy_toolchain",
```
Comment (fmeum) [inline MODULE.bazel:2]: We should consider renaming this to `toolchains_buildbuddy`. It's not a hard requirement, but calling a module `toolchains_X` if its purpose is to register toolchains for types it doesn't define itself has become a popular pattern in the BCR.

File: MODULE.bazel [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
 )
 
 bazel_dep(name = "rules_cc", version = "0.0.9")
 bazel_dep(name = "platforms", version = "0.0.10")
+
+buildbuddy = use_extension("//:extensions.bzl", "buildbuddy")
+use_repo(buildbuddy, "buildbuddy_toolchains")
+
+register_toolchains(
+    "@buildbuddy_toolchains//:ubuntu_cc_toolchain",
```
Comment (fmeum) [inline MODULE.bazel:14]: I find the plural s a bit too hard to spot here. Maybe call it `buildbuddy_toolchain_definitions` (or `toolchains_buildbuddy_definitions`)?

Comment (sluongng) [reply MODULE.bazel:14]: what if we do:
- Module: toolchains_buildbuddy
- Repo: buildbuddy_toolchain

🤔

File: MODULE.bazel [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+    "//toolchains/cc:ubuntu_gcc",
+    "//toolchains/cc:windows_msvc",
+)
+
+register_execution_platforms(
+    "//platforms:linux_x86_64",
+    "//platforms:linux_arm64",
+    "//platforms:darwin_x86_64",
+    "//platforms:darwin_arm64",
+    "//platforms:windows_x86_64",
```
Comment (fmeum) [inline MODULE.bazel:23]:
```suggestion
    "//platforms:all",
```
More maintainable in the long run.

File: MODULE.bazel [hunk 4]
```diff
@@ -1,8 +1,24 @@
 module(
-    name = "buildbuddy_toolchain",
-    version = "0.0.0",
+    name = "toolchains_buildbuddy",
+    version = "0.0.1",
```
Comment (fmeum) [inline MODULE.bazel:3]: If we plan to use Publish to BCR, then we don't need to bump this.

File: MODULE.bazel [hunk 5]
```diff
@@ -1,8 +1,24 @@
...
 
 bazel_dep(name = "rules_cc", version = "0.0.9")
 bazel_dep(name = "platforms", version = "0.0.10")
+
+buildbuddy = use_extension("//:extensions.bzl", "buildbuddy")
+use_repo(buildbuddy, "buildbuddy_toolchain")
+
+register_toolchains(
+    "//toolchains/cc:ubuntu_gcc",
+    "//toolchains/cc:windows_msvc",
```
Comment (fmeum) [inline MODULE.bazel:15]:
```suggestion
    "//toolchains/cc:all",
```
It's possible that aliases aren't supported with wildcards, in that case you could reference the unaliased package here.

File: examples/bzlmod/MODULE.bazel [hunk 6]
```diff
@@ -32,4 +35,7 @@ buildbuddy = use_extension("@buildbuddy//:extensions.bzl", "buildbuddy")
 #     )
 
 # All relevant toolchain and platform targets should be available under "@buildbuddy_toolchain//..." repository
+#
+# For convenience, the platform and toolchain targets are also "alias" in the module under
```
Comment (fmeum) [inline examples/bzlmod/MODULE.bazel:39]:
```suggestion
# For convenience, the platform and toolchain targets are also "alias"ed in the module under
```

**Review by fmeum — APPROVED** [review]

File: examples/bzlmod/MODULE.bazel [hunk 7]
```diff
@@ -32,4 +35,7 @@ buildbuddy = use_extension("@buildbuddy//:extensions.bzl", "buildbuddy")
...
+# "@toolchains_buildbuddy//platforms:all" and "@toolchains_buildbuddy//toolchains/cc:all" packages.
```
Comment (fmeum) [inline examples/bzlmod/MODULE.bazel:40]: Should we recommend those as the primary location? Then we wouldn't have to tell people about the extension repo and would be free to change its structure in the future. Users that migrate from WORKSPACE can already rename the repo with `repo_name`.

File: MODULE.bazel (hunk 4 above)
Comment (sluongng) [reply MODULE.bazel:3]: I want to publish the first version manually, and then use "Publish to BCR" workflows in the future. WDYT?

File: MODULE.bazel (hunk 5 above)
Comment (sluongng) [reply MODULE.bazel:15]: Yeah wild card does not work. Gona use the alias targets instead.

File: examples/bzlmod/MODULE.bazel (hunk 7 above)
Comment (fmeum) [reply examples/bzlmod/MODULE.bazel]: Users could still bring in the extension repo under an arbitrary name with `use_repo`.

What do we gain by advertising both the module repo and the extension repo though? It seems more difficult to maintain both, so how about either just advertising the `alias` targets (don't tell users about the `use_repo`) or doing away with the `alias` targets (which requires committing to the extension repo name)?

File: MODULE.bazel (hunk 4 above)
Comment (fmeum) [reply MODULE.bazel:3]: That's fine. I've messed up most manual releases in the past, but the app is also not exactly forgiving. We'll figure it out :-)

File: README.md [hunk 8]
```diff
@@ -4,6 +4,26 @@ Currently supports Linux C/C++ (including CGO) & Java builds on Ubuntu 16.04 or
...
+#### BzlMod
+
+Add the following to your `MODULE.bazel` file:
+
+```python
+bazel_dep(name = "toolchains_buildbuddy")
+
+buildbuddy = use_extension("@buildbuddy//:extensions.bzl", "buildbuddy")
+
+use_repo(buildbuddy, "buildbuddy_toolchain")
```
Comment (fmeum) [inline README.md:18]: Should we recommend this by default? It doesn't do anything by itself unless users would commonly want to reference the `buildbuddy_toolchain` repo.

**Review by fmeum — APPROVED** [review]

File: README.md (hunk 8 above)
Comment (sluongng) [reply README.md:18]: That's a good call. Now that we have the alias in place, we can just use the repo as a transitive dep instead of a direct dep.

**Pull request merged**